app/src/main/java/com/example/sharewhatyoucanproject/home/HomeViewModel.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.sharewhatyoucanproject.home
+
+import androidx.lifecycle.ViewModel
+import com.google.firebase.auth.FirebaseAuth
+
+class HomeViewModel(): ViewModel() {
+
+    private val currentUser = FirebaseAuth.getInstance().currentUser
```

Comment:
Putting this here means that it will be set once and never change, so the current user could become null but the view model would not reflect that change. You could write it this way so that you are always getting the most up to date value:

```
private val currentUser: FirebaseUser // not sure what the class name is here
   get() = FirebaseAuth.getInstance().currentUser
```

Reply:
done